### ISSUE TITLE
[Insight] Database queries should use parameter binding - in src/…/Repository/GlycemieRepository.php, line 20

### ISSUE THREAD

**Issue by artandor** [artandor/diabetehelper#4]
**[in src/DiabeteHelperBundle/Repository/GlycemieRepository.php, line 20](https://github.com/artandor/diabetehelper/blob/d3d26e1328a4508f4d579705a7c51daacc7dc719/src/DiabeteHelperBundle/Repository/GlycemieRepository.php#L20)**

> If provided by the user, the value of `$idUser` may allow an SQL injection attack. Avoid concatenating parameters to SQL query strings, and use parameter binding instead.

```php
    //dump($dateStart, $dateEnd);
    return $this->getEntityManager()
      ->createQuery(
        "
SELECT g FROM DiabeteHelperBundle:Glycemie g 
WHERE g.iduser = ". $idUser ." AND g.dateGlycemie BETWEEN '". $dateStart->format('Y-m-d') ."' AND '". $dateEnd->format('Y-m-d') ."' 
ORDER BY g.dateGlycemie DESC
"
      )
      ->getResult();
  }
```

---------------------

_Posted from [SensioLabsInsight](https://insight.sensiolabs.com/projects/ddfa8ed0-2ecc-4355-933c-6884d1599dc0/analyses/5?status=violations)_

**Comment by vasilvestre** [comment]
$qb = $this->createQueryBuilder('glycemie')
            ->andWhere('glycemie.idUser = :idUser')
            ->andWhere('glycemie.dateGlycemie BETWEEN :start AND :end')
            ->setParameter('idUser', $idUser)
            ->setParameter('start', $dateStart->format('Y-m-d'))
            ->setParameter('end', $dateEnd->format('Y-m-d'))
        ;
        
return $qb->getQuery()->execute();

Should do the job

**Comment by artandor** [comment]
Solved at commit 09d1f215b45c03e11903c7d9543db5733672f587 . Answer of @vasilvestre worked perfectly.
